samples: dfu: compressed_update: add TFM targets (#30075)
nrfconnect/sdk-nrf: doc/nrf/app_dev/bootloaders_dfu/mcuboot_image_compression.rst

    SB_CONFIG_BOOTLOADER_MCUBOOT=y
    SB_CONFIG_MCUBOOT_MODE_OVERWRITE_ONLY=y
    SB_CONFIG_MCUBOOT_COMPRESSED_IMAGE_SUPPORT=y

> 

Reviewer: Missing changelog entry

RAM usage with TF-M
*******************

> Avoiding memory leaks is important for security. Leaks can be avoided reusing MCUboot SRAM for secure application. 

Reviewer: ```suggestion
Avoiding memory leaks is important for security.
Reuse MCUboot SRAM for the secure application to reduce the risk of memory leaks.

```

Compressed image updates whith standard LZMA parameters are RAM intensive so allocating the same size for secure application results in very little left for non-secure

> application. To make it more viable sample configuration reduces dictionary size.

Reviewer: ```suggestion
Compressed image updates with standard LZMA parameters are RAM-intensive.
Allocating the same amount of RAM for the secure application leaves very little memory for the non-secure application.
To improve viability, the sample configuration reduces the dictionary size.
```

> Other method for avoiding potentially sensitive data leaks is to use RAM_CLEANUP.

Reviewer: ```suggestion

Another method to prevent potentially sensitive data leaks is to use ``RAM_CLEANUP``.
```

    SB_CONFIG_BOOTLOADER_MCUBOOT=y
    SB_CONFIG_MCUBOOT_MODE_OVERWRITE_ONLY=y
    SB_CONFIG_MCUBOOT_COMPRESSED_IMAGE_SUPPORT=y

RAM usage with TF-M
*******************

Avoiding memory leaks is important for security.
Reuse MCUboot SRAM for the secure application to reduce the risk of memory leaks.

Compressed image updates with standard LZMA parameters are RAM-intensive.
Allocating the same amount of RAM for the secure application leaves very little memory for the non-secure application.
To improve viability, the sample configuration reduces the dictionary size.

> Another method to prevent potentially sensitive data leaks is to use ``RAM_CLEANUP``.

Reviewer: ```suggestion
Use ``RAM_CLEANUP`` to prevent potential leaks of sensitive data.
```

    SB_CONFIG_BOOTLOADER_MCUBOOT=y
    SB_CONFIG_MCUBOOT_MODE_OVERWRITE_ONLY=y
    SB_CONFIG_MCUBOOT_COMPRESSED_IMAGE_SUPPORT=y

RAM usage with TF-M
*******************

Compressed image updates are RAM intensive.

> MCUboot must allocate sizable buffers to decompress firmware, and the footprint is noticeably larger than for a minimal MCUboot build without compression.

Reviewer: ```suggestion
Compressed image updates require significant RAM.
MCUboot must allocate large buffers to decompress the firmware.
As a result, the memory footprint is noticeably larger than that of a minimal MCUboot build without compression.
```

When this feature is used together with TF-M, MCUboot will typically require more SRAM than the secure application partition (``sram0_s``) provides.

> In practice, the MCUboot devicetree configuration therefore often assigns MCUboot a larger SRAM region than TF-M and the secure application use after boot.

Reviewer: ```suggestion
When you use this feature together with TF-M, MCUboot typically requires more SRAM than the secure application partition (``sram0_s``) provides.
In practice, the MCUboot devicetree configuration often assigns a larger SRAM region to MCUboot than the one used by TF-M and the secure application after boot.
```

.. note::
   Enable :kconfig:option:`CONFIG_MCUBOOT_CLEANUP_RAM` for MCUboot TF-M builds so that SRAM is zeroed before the handoff to TF-M.

>    Without RAM cleanup, data written by MCUboot outside the secure application SRAM partition may remain in memory and could leak into the non-secure world.

Reviewer: ```suggestion
   Without RAM cleanup, data written by MCUboot outside the secure application SRAM partition can remain in memory and potentially leak into the non-secure world.
```
